service/controller/v12/resource/instance/create.go

```diff
@@ -131,6 +160,28 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		}
 	}
 
+	if hasResourceStatus(customObject, Stage, DeploymentCompleted) {
+		err := r.handleDeploymentCompletedStatus(ctx, customObject)
+		if IsDeploymentNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "deployment not found")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for creation")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+		} else if controllercontext.IsInvalidContext(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "missing dispatched output values in controller context")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not ensure deployment")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+		} else if blobclient.IsBlobNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "ignition blob not found")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+		}
+
```

Comment:
Is there possibly some useful log message that could be written here? I think with general aligning, we would in this case at least log `canceling resource` since resource execution prematurely ends here (it doesn't follow all the way to end). Does this make sense?

Reply:
I think that log happens only when `reconciliationcanceledcontext.SetCanceled(ctx)` is called which is not the case in this `if` branch.

Reply:
I put a generic error log just to be more verbose. That should `if` branch should be an edge case anyway.